nilearn/decomposition/tests/test_decomposition_estimators.py

```diff
@@ -318,13 +318,21 @@ def test_pass_masker_arg_to_estimator(
 
 
 @pytest.mark.timeout(0)
-@pytest.mark.parametrize("estimator", [CanICA, _MultiPCA, DictLearning])
-@pytest.mark.parametrize("data_type", ["nifti"])
-def test_nifti_maps_masker_(canica_data_single_img, estimator):
+@pytest.mark.parametrize("estimator", [CanICA, DictLearning])
+@pytest.mark.parametrize("data_type", ["nifti", "surface"])
+def test_nifti_maps_masker_(
+    data_type,  # noqa: ARG001
+    canica_data,
+    estimator,
+):
     """Check deprecation of nifti_maps_masker_."""
-    est = estimator()
+    est = estimator(
+        n_components=3,
+        random_state=RANDOM_STATE,
+        smoothing_fwhm=None,
+    )
 
-    est.fit(canica_data_single_img)
+    est.fit(canica_data)
```

Comment:
I am using the same set up as for other tests in this file that are not flaky: not quite sure what the source of the flakyness was but I would suspect using canica_data_single_img not being appropriate for DictLearning